velox/functions/sparksql/String.h

```diff
@@ -583,4 +583,83 @@ struct OverlayVarbinaryFunction : public OverlayFunctionBase {
   }
 };
 
+/// translate(string, matchStr, replaceStr) -> varchar
+///
+///   Returns a new translated string. It translates the character in ``string``
+///   by a character in ``replaceStr``. The character in ``replaceStr`` is
+///   corresponding to the character in ``matchStr``. The translation will
+///   happen when any character in ``string`` matching with the character in the
+///   ``matchStr``.
+template <typename T>
+struct TranslateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  using MapType = std::unordered_map<StringView, StringView>;
+  MapType dict_;
+  bool isConstantTranslateDict = false;
+
+  FOLLY_ALWAYS_INLINE MapType buildDict(
```

Comment:
do not abbreviate: buildDictionary, match, replace. Please, check out coding style guide and update.